blog/k8s-probe.md

```diff
@@ -0,0 +1,305 @@
+---
+title: Introducing the Kubernetes Probe
+layout: blog-post
+author: Aidan Shribman
+date: 24/04/2019
+---
+
+Since Skydive v0.22.0 the K8s topology probe is production ready. In this blog
+post I present the probes' capabilities and potential use cases.
+
+## Motivation
+
+K8s configuration is declarative, consisting of many resources such as the
+below network policy configuration (a default deny all rule):
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: deny
+spec:
+  podSelector: {}
+  policyTypes:
+  - Egress
+```
+
+Tools such as `kubectl` extract the declarative configuration 
+(plus extra status fields) which is fine, but lacking in  terms of
+being able to understand how the network policy resource interacts with other
+resources, such as with the below given pods:
+
+```
+apiVersion: core/v1
+kind: Pod
+metadata:
+  name: a
+spec:
+  ...
+---
+apiVersion: core/v1
+kind: Pod
+metadata:
+  name: b
+spec:
+  ...
+```
+
+Even answering a trivial question such as if traffic is allowed to flow from
+`pod:a` to `pod:b` may not be easy.
+
+In fact traffic is denied due to `networkpolicy:deny` governing both `pod:a`
+and `pod:b` as all reside in the default namespace and the `networkpolicy:deny`
+pod-selector is empty (indicating it applies to all pods in the namespace).
+
+A graph such as the below would greatly enhance the understanding of how the
+`networkpolicy:deny` influences `pod:a` and `pod:b`:
+
+```
+pod:a <---> networkpolicy:deny <---> pod:b
+```
+
+More generally K8s may define relations between resources in many ways such as
+by:
+- namespace/name
+- label-selectors
+- pod-selectors
+- namespace-selectors
+- ip-blocks
+
+The Skydive k8s probe is able to understand all these relations and construct
+graph edges enabling to more easily reason about your k8s landscape.
+
+## Getting Started
+
+If you don't already have a k8s cluster you can setup a minimal single node k8s
```

Comment:
If you don't already have a Kubernetes cluster yet, you can setup a minimal single node cluster for evaluation purposes using `minikube`

Reply:
done